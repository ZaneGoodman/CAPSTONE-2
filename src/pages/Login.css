.Login {
  text-align: center;
  margin: 40px;
}

.Login-header {
  color: rgb(236, 234, 226);
  font-family: "Rosarivo", cursive;
  text-align: center;
}

.Login-check {
  color: rgb(236, 234, 226);

  font-family: "Rosarivo", cursive;
  text-align: center;
}
.Login-link {
  text-decoration: none;
  color: rgb(236, 234, 226);
  cursor: pointer;
}
.Login-background {
  background-image: url("https://cdn.pixabay.com/photo/2014/10/22/18/16/church-498525_1280.jpg");

  background-size: cover;
  background-repeat: no-repeat;
  position: absolute;
  margin: 0;
  padding: 0;
  left: 0;
  top: 49px;
  bottom: 0;
  right: 0;
  overflow: hidden;
}
