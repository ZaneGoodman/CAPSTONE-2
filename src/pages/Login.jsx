import React, { useState } from "react";
import { useAuth } from "../provider/authProvider";
import { useNavigate, Link } from "react-router-dom";
import AuthForm from "../forms/AuthForm";
import Authorization from "../models/authUser";

const Login = () => {
  const { setToken, setUsername } = useAuth();

  const navigate = useNavigate();

  const INITIAL_STATE = {
    username: "",
    password: "",
  };
  const [userData, setUserData] = useState(INITIAL_STATE);

  const handleLogin = async (fData) => {
    setUserData(() => fData);
    setToken(await Authorization.login(fData.username, fData.password));
    setUsername(fData.username);
    navigate("/", { replace: true });
    navigate(0);
  };

  return (
    <>
      <h1>Welcome to the Prayer Room </h1>
      <h2>Come pray the rosary with us</h2>
      <AuthForm login={handleLogin} />
      <span>
        Dont have an account? <Link to={"/signup"}>Sign up</Link>
      </span>
    </>
  );
};

export default Login;
