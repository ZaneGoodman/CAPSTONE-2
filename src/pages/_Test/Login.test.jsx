import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import AuthProvider from "../../provider/authProvider";
import Login from "../Login";
//Smoke test
test("renders Login component with authentication", async () => {
  render(
    <AuthProvider initalState={{ token: "testToken", username: "bob" }}>
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    </AuthProvider>
  );
});

//Snapshot test
it("Login matches snapshot", async () => {
  const { asFragment } = render(
    <AuthProvider initalState={{ token: "testToken", username: "bob" }}>
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    </AuthProvider>
  );
  await expect(asFragment()).toMatchSnapshot();
});
