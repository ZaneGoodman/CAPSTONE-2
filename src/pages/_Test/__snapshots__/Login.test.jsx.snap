// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    Welcome to the Prayer Room 
  </h1>
  <h2>
    Come pray the rosary with us
  </h2>
  <form>
    <label
      for="username"
    >
      Username
    </label>
    <input
      id="username"
      name="username"
      value=""
    />
    <label>
      Password
    </label>
    <input
      id="password"
      name="password"
      value=""
    />
    <button>
      Go!
    </button>
  </form>
  <span>
    Dont have an account? 
    <a
      href="/signup"
    >
      Sign up
    </a>
  </span>
</DocumentFragment>
`;
